### PR TITLE
Debug Normals

### PR DESCRIPTION
**Pull request by jurek1029**

### CODE REVIEW

File: PolyEngine/Engine/Res/debugVertSh.shader [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+out VS_OUT {
+    vec3 normal;
+} vs_out;
+
+uniform mat4 projection;
+uniform mat4 viewModel;
+
+void main()
+{
+	gl_Position = projection * viewModel * vec4(position, 1.0f);
```
Comment (MuniuDev) [inline PolyEngine/Engine/Res/debugVertSh.shader]: Move `projection * viewModel` to host and pass mvp as single uniform.

Also i suggest using "u_" for uniforms, "a_" for attributes anf "v_" for varyings (in/out)

File: PolyEngine/Engine/Src/CoreConfig.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+namespace Poly
+{
+
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: unnecessary empty line

File: PolyEngine/Engine/Src/RenderingSystem.cpp [hunk 3]
```diff
@@ -51,6 +51,30 @@ void RenderingSystem::RenderingPhase(World* world)
 				glBindVertexArray(0);
 			}
 		}
+		if (gCoreConfig.DebugNormalsFlag) 
+		{
+			const Matrix& mModelView = kv.second.GetCamera()->GetModelViewMatrix();
```
Comment (MuniuDev) [inline PolyEngine/Engine/Src/RenderingSystem.cpp]: You can just call `GetMVP()` here as mentioned in previous comment.

File: PolyEngine/Engine/Res/debugVertSh.shader (hunk 1 above)
Comment (jurek1029) [inline PolyEngine/Engine/Res/debugVertSh.shader]: Done

File: PolyEngine/Engine/Src/CoreConfig.hpp (hunk 2 above)
Comment (jurek1029) [inline PolyEngine/Engine/Src/CoreConfig.hpp]: Gone

File: PolyEngine/Engine/Src/RenderingSystem.cpp (hunk 3 above)
Comment (jurek1029) [inline PolyEngine/Engine/Src/RenderingSystem.cpp]: you don't have to you, can get mvp from mvp variable in same scope

**Pull request merged**